modules/modules.go

```diff
@@ -184,6 +236,106 @@ func WatchForStop(r io.Reader, stopChan *chan bool) error {
 	}
 }
 
+func DefaultPersistHandlers(in io.ReadCloser, out io.WriteCloser, logch chan string) {
+	inChan := make(chan Message, 0)
+	go func() {
+		for {
+			msg, err := ReadInput(in)
+			if err != nil {
+				close(inChan)
+				break
+			}
+			inChan <- msg
+		}
+	}()
+	for {
+		failed := false
+
+		select {
+		case s := <-logch:
+			logmsg, err := MakeMessageLog("%v", s)
+			if err != nil {
+				failed = true
+				break
+			}
+			err = WriteOutput(logmsg, out)
+			if err != nil {
+				failed = true
+				break
+			}
+		case msg, ok := <-inChan:
+			if !ok {
+				failed = true
+				break
+			}
+			switch msg.Class {
+			case "ping":
+				buf, err := json.Marshal(&msg)
+				if err != nil {
+					failed = true
+					break
+				}
+				err = WriteOutput(buf, out)
+				if err != nil {
+					failed = true
+					break
+				}
+			}
+		}
+
+		if failed {
+			break
+		}
+	}
+}
+
+func HandlePersistRequest(l net.Listener, f func(interface{}) string) {
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			panic(err)
+		}
+		go func() {
+			var p interface{}
+			err = ReadInputParameters(conn, &p)
+			if err != nil {
+				panic(err)
+			}
+			resp := f(p)
+			WriteOutput([]byte(resp), conn)
+			err = conn.Close()
+			if err != nil {
+				panic(err)
+			}
+		}()
+	}
+}
+
+func SendPersistRequest(p interface{}, modname string) string {
+	conn, err := net.Dial("unix", PersistSockPath(modname))
+	if err != nil {
+		panic(err)
+	}
+	buf, err := MakeMessage(MsgClassParameters, p, false)
+	if err != nil {
+		panic(err)
+	}
+	err = WriteOutput(buf, conn)
+	if err != nil {
+		panic(err)
+	}
+	rb, err := ioutil.ReadAll(conn)
+	if err != nil {
+		panic(err)
+	}
+	return string(rb)
+}
+
+func PersistSockPath(modname string) string {
```

Comment:
Consider registering and mapping the name to both the net and address parameters so that when looking up by name both the net and address parameters can be returned and passed to Listen and Dial for supporting tcp connections on Windows (if needed).